examples/platforms/da15000/platform.c

```diff
@@ -0,0 +1,166 @@
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * This file includes the platform-specific initializers.
+ */
+
+#include <assert.h>
+#include <stdint.h>
+
+#include <openthread.h>
+#include "platform/alarm.h"
+#include <openthread-types.h>
+#include "platform-da15000.h"
+#include "black_orca.h"
+#include "hw_cpm.h"
+#include "hw_watchdog.h"
+#include "ftdf.h"
+#include "hw_gpio.h"
+#include "platform/uart.h"
+
+uint32_t NODE_ID = 1;
+
+static bool blink = false;
+static int  mscounter_init;
+static int  mscounter;
+
+otInstance *sInstance;
+
+
+void ClkInit(void)
+{
+
+    NVIC_ClearPendingIRQ(XTAL16RDY_IRQn);
+    NVIC_EnableIRQ(XTAL16RDY_IRQn);                 // Activate XTAL16 Ready IRQ
+    hw_cpm_set_divn(false);                         // External crystal is 16MHz
+    hw_cpm_enable_rc32k();
+    hw_cpm_lp_set_rc32k();
+    hw_cpm_set_xtal16m_settling_time(dg_configXTAL16_SETTLE_TIME_RC32K);
+    hw_cpm_enable_xtal16m();                        // Enable XTAL16M
+    hw_cpm_configure_xtal32k_pins();                // Configure XTAL32K pins
+    hw_cpm_configure_xtal32k();                     // Configure XTAL32K
+    hw_cpm_enable_xtal32k();                        // Enable XTAL32K
+    hw_watchdog_unfreeze();                         // Start watchdog
+
+    while (!hw_cpm_is_xtal16m_started());           // Block until XTAL16M starts
+
+    hw_watchdog_freeze();                           // Stop watchdog
+    hw_cpm_set_recharge_period((uint16_t)dg_configSET_RECHARGE_PERIOD);
+    hw_cpm_set_sysclk(SYS_CLK_IS_XTAL16M);
+    hw_cpm_set_hclk_div(0);
+    hw_cpm_set_pclk_div(0);
+}
+
+/*
+ * Example function. Blink LED according to node state
+ * Leader       - 5Hz
+ * Router       - 2Hz
+ * Child        - 0.5Hz
+ */
+
+void ExampleProcess(otInstance *aInstance)
+{
+    otDeviceRole DevRole;
+
+    DevRole = otGetDeviceRole(aInstance);
+
+    if (blink == false && otPlatAlarmGetNow() > 0)
+    {
+        mscounter_init = otPlatAlarmGetNow();
```

Comment:
otPlatAlarmGetNow() is a "get" function for a global variable. Compiler will not perform subroutine jump, it will inline global variable. Current solution is readable and resource-friendly.